relay-server/src/services/processor/span/processing.rs

```diff
@@ -414,12 +414,15 @@ fn get_normalize_span_config<'a>(
 fn set_segment_attributes(span: &mut Annotated<Span>) {
     let Some(span) = span.value_mut() else { return };
 
-    // TODO: A span might be a segment span even if the parent_id is not empty
-    // (parent within a trace). I.e. do not overwrite here.
-    let is_segment = span.parent_span_id.is_empty();
+    let is_root = span.parent_span_id.is_empty();
 
-    span.is_segment = Annotated::new(is_segment);
-    if is_segment {
+    // If `is_segment` was not set by the SDK and the span has no parent, assume it is a segment:
+    if is_root && span.is_segment.value().is_none() {
+        span.is_segment = true.into();
+    }
+
+    // If the span is a segment, always set to segment_id to the current span_id:
+    if span.is_segment.value() == Some(&true) {
         span.segment_id = span.span_id.clone();
     }
```

Comment:
Shouldn't we also check for this, if the SDK does that already we have to set `is_segment` to true as well.

Reply:
You mean check the opposite direction? I.e. set `is_segment` if `segment_id == span_id`?

Reply:
Expand `is_root` to:

```
let is_root = span.parent_span_id.is_empty() || span.segment_id == span.span_id;
```

Reply:
Added this now. I also changed the logic to _not_ accept `is_segment: false` from the SDK if the condition above is met.